starknet_py/tests/e2e/account/account_client_test.py

```diff
@@ -51,3 +58,60 @@ async def test_error_when_tx_signed(run_devnet):
     assert "Adding signatures to a signer tx currently isn't supported" in str(
         t_err.value
     )
+
+
+@pytest.mark.asyncio
+async def test_error_when_token_not_specified(run_devnet):
+    acc_client = await AccountClient.create_account(
+        net=run_devnet, chain=StarknetChainId.TESTNET
+    )
+
+    with pytest.raises(BadRequest) as b_req:
+        await acc_client.get_balance()
+
+    assert "Specify token_address for custom url." in str(b_req.value)
+
+
+@pytest.mark.asyncio
+async def test_balance_when_token_specified(run_devnet):
+    acc_client = await AccountClient.create_account(
+        net=run_devnet, chain=StarknetChainId.TESTNET
+    )
+
+    deployment_result = await Contract.deploy(
+        client=acc_client, compilation_source=erc20_mock_source_code
+    )
+    deployment_result = await deployment_result.wait_for_acceptance()
+    erc20_contract = deployment_result.deployed_contract
+
+    balance = await acc_client.get_balance(erc20_contract.address)
+
+    assert balance == 200
+
+
+@pytest.mark.asyncio
+async def test_default_token_address():
+
+    acc_client_testnet = AccountClient("0x123", KeyPair(123, 456), TESTNET)
+    acc_client_mainnet = AccountClient("0x321", KeyPair(456, 123), MAINNET)
+
```

Comment:
Would it be possible to `@pytest.mark.parametrize` this test with respective clients and expected eth contract address? Now we are testing two accounts at the same time and it makes the test a little bit confusing